### ISSUE TITLE
how do you know "com.apple.wifi.linkdidchange"

### ISSUE THREAD

**Issue by powfulhong** [davidmurray/wifi#22]
hi, is there any "com.apple.wifi.***" that shows wifi's state? I found that after the wifi's association   succeed， wifi icon would wait for seconds to show in the status bar...how do i know its truly time that everything is ready? wait for your answer...(poor english, sorry)

**Comment by davidmurray** [comment]
You mean a notification that is sent when WiFi turns on or turns off?

**Comment by powfulhong** [comment]
No, just like when android wifi is associating, it has a few states that would broadcast, for example authorization.what I want to know is what notification it will send when iOS wifi is associating. Because I found when association callback returns success, the wifi icon don't appear right now,likely it is doing something.

发自我的 iPad

**Comment by davidmurray** [comment]
Try `WiFiDeviceClientAssociateAsync` with the callback argument.

**Comment by powfulhong** [comment]
I did use WiFiDeviceClientAssociateAsync and the callback show wifi was associating successfully . What i confused is why it would wait a few seconds to appear at top-left. And how do you know there is a notification called "com.apple.wifi.linkdidchange"? How can i find others? Thanks.

**Comment by davidmurray** [comment]
Run the command `strings` on a MobileWiFi binary. You should see all the notifications.

**Comment by powfulhong** [comment]
I try  strings MobileWiFi | grep "com.apple", but it just return ------
com.apple.wifi.manager.available
com.apple.wifi.manager
com.apple.wifid.queue
com.apple.wifi.manager.available
com.apple.wifi.manager
com.apple.wifid.queue
com.apple.wifi.manager.available
com.apple.wifi.manager

## com.apple.wifid.queue

is it right???(Though i do think so...)

**Comment by powfulhong** [comment]
or can you help me find out all notifications of MobileWiFi?

**Comment by davidmurray** [comment]
Sorry, not sure how you can find the other notifications. :( Maybe check the `wifid` binary?

**Comment by powfulhong** [comment]
can't find the other notifications.....:(   i found when the association callback return success, it was actually doing something, for example, getting the ip address.  
now i found another problem that even if i enter the wrong pwd, the association would return success but actually associated to a different wifi? How could it be?

**Comment by powfulhong** [comment]
And how did you get *.h???  class_dump doesn't get it. I use nm MobileWiFi and find that there are a lot of 'information' that you didn't show in headers...

**Comment by davidmurray** [comment]
I'm not really sure about the other issue you're experiencing. I don't know everything about the iOS WiFi.

I didn't use class-dump, I manually reverse engineered the functions.
